services/catalog/src/simcore_service_catalog/services/services_api.py

```diff
@@ -110,13 +123,24 @@ async def list_services_paginated(
         (s.key, s.version): s for s in got if isinstance(s, ServiceMetaDataPublished)
     }
 
-    # NOTE: aggregates published (i.e. not editable) is still missing in this version
     items = [
-        _db_to_api_model(s, ar, sm)
+        _db_to_api_model(
+            service_db=s, access_rights_db=ar, service_manifest=sm, compatibility_map=cm
+        )
         for s in services
         if (
             (ar := access_rights.get((s.key, s.version)))
             and (sm := service_manifest.get((s.key, s.version)))
+            and (
+                # NOTE: This operation might be resource-intensive.
+                # It is temporarily implemented on a trial basis.
+                cm := await evaluate_service_compatibility_map(
```

Comment:
Q: could you not parallelise these and put them back together? How big do you expect this items array to be? 1-10 items? 100 items?
What I'm trying to say, is that this could get rather slow.

Reply:
this is serving a paginated function meaning at most 20 items at a time